packages/backend/src/controllers/tasks/tasks.controller.ts

```diff
@@ -0,0 +1,275 @@
+            };
+          },
+        );
+        return {
+          tasks: updatedTasksDto,
+        };
+      }
+    }
+
+    throw new HttpException('user is not in a house', HttpStatus.BAD_REQUEST);
+  }
+
+  @Delete('/tasks/:id')
+  @HttpCode(204)
+  @ApiOperation({ summary: 'delete a task from a house.' })
+  @ApiNoContentResponse({
+    description: 'task successfully deleted.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not the owner of the house',
+  })
+  async deleteTaskFromHouse(@Param('id') id, @User() user: DecodedIdToken) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house) {
+      const house = await this.houseStoreService.findOne(userDoc.house);
+      const task = await this.taskStoreService.findOne(id);
+
+      if (house.owner.equals(userDoc._id)) {
+        this.taskStoreService.delete(task._id);
+        return 'task deleted successfully';
+      }
+
+      throw new HttpException(
+        'user is not the owner of the house',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    throw new HttpException('user is not in the house', HttpStatus.BAD_REQUEST);
+  }
+
+  @Put('/task/:id/completed')
+  @ApiOperation({
+    summary: 'mark user as having completed/not completed task.',
+  })
+  @ApiResponse({
+    description: 'tasks successfully marked as complete/not complete.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not assigned to the task',
+  })
+  @ApiNotFoundResponse({
+    description: 'task not found',
+  })
+  async markTaskAsComplete(
+    @Body() completeTaskDto: CompleteTaskDto,
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+  ) {
+    if (!isValidObjectId(id)) {
+      throw new HttpException('task does not exist', HttpStatus.NOT_FOUND);
+    }
+    const task = await this.taskStoreService.findOne(id);
+
+    if (task) {
+      if (task.assigned !== user.uid) {
+        throw new HttpException(
+          'user is not assigned to task',
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+
+      if (completeTaskDto.isComplete) {
+        this.taskStoreService.update(task._id, { lastCompleted: new Date() });
+      } else {
+        this.taskStoreService.update(task._id, { lastCompleted: null });
+      }
+      return;
+    }
+
+    throw new HttpException('task not found', HttpStatus.NOT_FOUND);
+  }
+
+  @Put('/task/:id')
+  @ApiOperation({ summary: 'Modify task name, decription or pool' })
+  @ApiResponse({
+    description: 'task successfuly updated.',
+  })
+  @ApiBadRequestResponse({
+    description: 'user is not the owner of the house or not in the house',
+  })
+  @ApiBadRequestResponse({
+    description: 'task does not exist',
+  })
+  async modifyTask(
+    @Param('id') id,
+    @User() user: DecodedIdToken,
+    @Body() updateHouseTasksDto: UpdateHouseTasksDto,
+  ) {
+    const userDoc = await this.userStoreService.findOneByFirebaseId(user.uid);
+
+    if (userDoc.house) {
```

Comment:
Inversing the condition and immediately throwing the exception here is more readable and avoids nesting statements.